osu.Game/Screens/Play/HUD/MultiplayerGameplayLeaderboard.cs

```diff
@@ -80,6 +78,8 @@ protected override void LoadComplete()
             // BindableList handles binding in a really bad way (Clear then AddRange) so we need to do this manually..
             foreach (int userId in playingUsers)
             {
+                spectatorClient.WatchUser(userId);
```

Comment:
Curious about the moving of this call. Is there a reason it needs to be run from the update thread to guarantee behaviour? Or just felt like a better place for it to sit?

Reply:
The other way is to schedule internally to `WatchUser`, otherwise it leads to potential cross-thread mutation of a list.

Reply:
Do we not want a bit more safety here? Can we use `ThreadSafety` checks at least?

Reply:
Will add an assert, sure.

Reply:
Apart from this it looks fine. I'd probably add asserts anywhere the safety is required to play it safe.